04.ls/ls.rb

```diff
@@ -33,4 +33,113 @@ def split_files(files)
   divided_files
 end
 
+PERMISSION = {
+  '0' => '---',
+  '1' => '--x',
+  '2' => '-w-',
+  '3' => '-wx',
+  '4' => 'r--',
+  '5' => 'r-x',
+  '6' => 'rw-',
+  '7' => 'rwx',
+  'fifo' => 'p',
+  'characterSpecial' => 'c',
+  'directory' => 'd',
+  'blockSpecial' => 'b',
+  'file' => '-',
+  'link' => 'l',
+  'socket' => 's'
+}.freeze
+
+def load_l_option(path, file_names)
+  stat_files = file_names.map { |faile_name| File.lstat("#{path}/#{faile_name}") }
+  path_to_stat = stat_files.map.with_index { |stat, idx| ["#{path}/#{file_names.fetch(idx)}", stat] }.to_h
+
+  puts "total #{stat_files.map(&:blocks).sum}"
+  load_body(path_to_stat).each do |file|
+    puts format_file(file, stat_files)
+  end
+end
+
+def load_body(path_to_stat)
+  path_to_stat.map do |absolute_path, stat|
+    {
+      permission: to_permission(stat),
+      link_count: stat.nlink.to_s,
+      owner: Etc.getpwuid(stat.uid).name,
+      group: Etc.getgrgid(stat.gid).name,
+      byte_size: stat.size.to_s,
+      time_stamp: stat.mtime.strftime(
+        Time.now.year == stat.mtime.year ? ' %_m %_d %H:%M ' : ' %_m %_d  %Y '
+      ),
+      file_name:
+        if stat.symlink?
+          "#{File.basename(absolute_path)} -> #{File.readlink(absolute_path)}"
+        else
+          File.basename(absolute_path)
+        end
+    }
+  end
+end
+
+def format_file(file, stat_files)
+  block_to_maxlength = fetch_max_length(stat_files)
+  constant_size = 2
+  [
+    file.fetch(:permission),
```

Comment:
```suggestion
    file[:permission],
```

とした方がすっきり書けると思うのですがどうでしょうか？

Reply:
`file[ ]`のようにファイル全体を修正しました。
記法自体は知っていたのですが、`fetch`を使った方がどこかしらでデータが欠けている時にエラーの発生箇所がわかりやすいかなと思い`fetch`を使っていました。

Reply:
> fetchを使った方がどこかしらでデータが欠けている時にエラーの発生箇所がわかりやすいかなと思いfetchを使っていました。

なるほど、たしかにそういう発想もありかもしれませんね。
ただ、このプログラムの場合、「どこかしらでデータが欠けている時」というのはただのバグだと思うので、しっかりテストすれば`[]`でも問題ないはずです。
`fetch`を使うときはたとえば、外部から渡されたデータでそのキーが本当に存在しているかどうか確信がない、みたいなときに使うといいかな〜と思います。（そうしないと、`fetch`だらけになってコードが読みづらくなってしまう）